src/api/java/com/minecolonies/api/crafting/ItemStorage.java

```diff
@@ -103,6 +109,55 @@ public ItemStorage(@NotNull final ItemStack stack)
         this.creativeTabIndex = stack.getItem().getCreativeTabs().stream().filter(Objects::nonNull).map(g -> g.index).collect(Collectors.toList());
     }
 
+    /**
+     * Creates an instance of the storage from JSON
+     * 
+     * @param jObject the JSON Object to parse
+     */
+    public ItemStorage(@NotNull final JsonObject jObject)
+    {
+        if (jObject.has(ITEM_PROP))
+        {
+            final ItemStack parsedStack = ItemStackUtils.idToItemStack(jObject.get(ITEM_PROP).getAsString());
+            if(jObject.has(COUNT_PROP))
+            {
+                parsedStack.setCount(jObject.get(COUNT_PROP).getAsInt());
+                this.amount = jObject.get(COUNT_PROP).getAsInt();
+            }
+            else
+            {
+                this.amount = parsedStack.getCount();
+            }
+            this.stack = parsedStack;
+            if(jObject.has(MATCHTYPE_PROP))
+            {
+                String matchType = jObject.get(MATCHTYPE_PROP).getAsString();
+                if(matchType.equals(MATCH_NBTIGNORE))
+                {
+                    this.shouldIgnoreNBTValue = true;
+                }
+                else // includes "exact"
+                {
+                    this.shouldIgnoreNBTValue = false;
+                }
```

Comment:
We're using this. for all of the constructor assigns,  and while I could assign the result of the .equals, this is the structure it will become when we add additional options for matchtype.